### PR TITLE
Add IBM Power VS: manifest assets

### PR DESCRIPTION
**Pull request by clnperez**
For more background on IPI on Power VS, refer to the enhancement
proposal here: openshift/enhancements#736

Older discussions on some of the code here can be found in
https://github.com/openshift/installer/pull/5224

Signed-off-by: Christy Norman <christy@linux.vnet.ibm.com>

### CODE REVIEW

File: pkg/asset/manifests/infrastructure.go [hunk 1]
```diff
@@ -211,6 +212,17 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
 			APIServerInternalIP: installConfig.Config.Ovirt.APIVIP,
 			IngressIP:           installConfig.Config.Ovirt.IngressVIP,
 		}
+	case powervs.Name:
+		config.Spec.PlatformSpec.Type = configv1.PowerVSPlatformType
+		cisInstanceCRN, err := installConfig.PowerVS.CISInstanceCRN(context.TODO())
+		if err != nil {
+			return err
```
Comment (rna-afk) [inline pkg/asset/manifests/infrastructure.go:219]:
```suggestion
			return errors.Wrapf(err, "failed to get instance CRN")
```

**Review by kirankt — COMMENTED** [review]
Before you commit changes to the PR, please make sure that you've run the linters and tests. This will help cut down failed CI tests.
https://github.com/openshift/installer/blob/master/CONTRIBUTING.md#contribution-flow

**Comment by clnperez** [comment]
@kirankt yep, thanks.i thought we had all those covered in our fork where we've been doing dev for the past several months but it looks like we missed some. will address those for sure

**Comment by clnperez** [comment]
I think we'll need to get at least https://github.com/openshift/installer/pull/5609 merged before this will build and really run any tests

File: pkg/asset/manifests/dns.go [hunk 2]
```diff
@@ -158,7 +159,7 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 		config.Spec.PrivateZone = &configv1.DNSZone{
 			ID: zoneID,
 		}
-	case libvirttypes.Name, openstacktypes.Name, baremetaltypes.Name, nonetypes.Name, vspheretypes.Name, ovirttypes.Name:
+	case libvirttypes.Name, openstacktypes.Name, baremetaltypes.Name, nonetypes.Name, vspheretypes.Name, ovirttypes.Name, powervstypes.Name:
```
Comment (Prashanth684) [inline pkg/asset/manifests/dns.go:162]: wouldn't this be the same as ibmcloud above?

Comment (clnperez) [reply pkg/asset/manifests/dns.go:162]: @Karthik-K-N -- can you answer this? i can't tell from looking at your Power VS support pr here https://github.com/openshift/cluster-ingress-operator/pull/658/files but am thinking we may be able to just add the powervs case to the ibmcloud case here (which is what i'm pretty sure Prashanth is suggesting)?

Comment (Karthik-K-N) [reply pkg/asset/manifests/dns.go:162]: Yes, We need to add PowerVS to IBMCloud case block and its tracked here https://issues.redhat.com/browse/MULTIARCH-1603 and also I think as per the slack discussion I see, @miyamotoh already done this.

**Comment by clnperez** [comment]
looks like this also depends on https://github.com/openshift/installer/pull/5612

**Comment by clnperez** [comment]
added cloud-config while we wait for the others, but this does now require https://github.com/openshift/installer/pull/5700

**Comment by clnperez** [comment]
i think this one is good to go now @patrickdillon @Prashanth684 @rna-afk

File: pkg/asset/manifests/powervs/cloudproviderconfig.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	Region                   string `gcfg:"region"`
+	G2CredentialsFilePath    string `gcfg:"g2Credentials"`
+	G2ResourceGroupName      string `gcfg:"g2ResourceGroupName"`
+	G2VPCName                string `gcfg:"g2VpcName"`
+	G2WorkerServiceAccountID string `gcfg:"g2workerServiceAccountID"`
+	G2VPCSubnetNames         string `gcfg:"g2VpcSubnetNames"`
+}
+
+// CloudProviderConfig generates the cloud provider config for the IBMCloud platform.
+func CloudProviderConfig(infraID string, accountID string, vpcName string, region string, resourceGroupName string, subnets []string) (string, error) {
```
Comment (Prashanth684) [inline pkg/asset/manifests/powervs/cloudproviderconfig.go:37]: can this not be common with ibmcloud? i don't see any difference other than the subnets?

Comment (clnperez) [reply pkg/asset/manifests/powervs/cloudproviderconfig.go:37]: i believe @Karthik-K-N said they'll be adding some additional items when they get the ccm fork available, so i'd defer to him on that. but if we do agree to that let's make a tracker so we don't misplace that thought. in the meantime, ok to merge this as-is?

Comment (Prashanth684) [reply pkg/asset/manifests/powervs/cloudproviderconfig.go:37]: if there's going to be some deviation then yeah, it's ok

Comment (Prashanth684) [reply pkg/asset/manifests/powervs/cloudproviderconfig.go:37]: also a nit to change the above comment from IBMCloud to PowerVS

**Pull request merged**